File: 백준/Silver/11723. 집합/집합.cc
#include <bits/stdc++.h>
#define endl '\n'
#define MAX 22
using namespace std;

int arr[MAX];

int main() {
	ios_base :: sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);
	
	int n, x;
	string str;
	cin >> n;
	for(int i=0; i<n; i++) {
		cin >> str;
		if(str=="all") {
			for(int j=1; j<=20; j++) {
				arr[j]=1;
			}
		} 
		else if(str=="empty") {
			for(int j=1; j<=20; j++) {
				arr[j]=0;
			}
		}
		else {
			cin >> x;	
			if(str=="add") {
				arr[x]=1;
			} else if(str=="remove") {
				arr[x]=0;
			} else if(str=="check") {
				if(arr[x]==1) cout << 1 << endl;
				else cout << 0 << endl;
			} else if(str=="toggle") {
				if(arr[x]==0) arr[x]=1;
				else arr[x]=0;
			}
		}
	}
	
	return 0;
}

File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int m = Integer.parseInt(st.nextToken());

        int[] S = new int[21];

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine(), " ");

            String command = st.nextToken();
            int x = 0;
            if (st.hasMoreTokens()) {
                x = Integer.parseInt(st.nextToken());
            }

            if (command.equals("add")) S[x] = 1;
            if (command.equals("remove")) S[x] = 0;
            if (command.equals("check")) sb.append(S[x] == 1 ? "1\n" : "0\n");
            if (command.equals("toggle")) {
                if (S[x] == 1) S[x] = 0;
                else S[x] = 1;
            }
            if (command.equals("all")) Arrays.fill(S, 1);
            if (command.equals("empty")) Arrays.fill(S, 0);
        }
        System.out.println(sb);
    }

}